datacube/utils/geometry/_base.py

```diff
@@ -972,66 +978,61 @@ def __eq__(self, other):
                 and self.crs == other.crs)
 
 
-def relative_translation(A: GeoBox, B: GeoBox):  # pylint: disable=invalid-name
+def bounding_box_in_pixel_domain(reference: GeoBox, other: GeoBox):
     """
-    Return the relative pixel translation `t` in
-    the equations `X_A` = `X_B` + `t`, where
-    the Xs are pixel coordinates of the same geographical point,
-    if their grids are compatible, otherwise return `None`.
+    Returns the bounding box of `other` with respect to the pixel grid
+    defined by `reference` when their coordinate grids are compatible,
+    otherwise raises `ValueError`.
```

Comment:
Worth spelling out what "compatible" means

- Same CRS
- Same pixel size and orientation
- Whole pixel translation (i.e. coincidental pixels)